test/node/double_tests.js

```diff
@@ -3,36 +3,96 @@
 const BSON = require('../register-bson');
 const Double = BSON.Double;
 
-describe('Double', function () {
-  describe('Constructor', function () {
-    var value = 42.3456;
+describe('BSON Double Precision', function () {
+  context('class Double', function () {
+    describe('constructor()', function () {
+      const value = 42.3456;
 
-    it('Primitive number', function (done) {
-      expect(new Double(value).valueOf()).to.equal(value);
-      done();
+      it('Primitive number', function (done) {
+        expect(new Double(value).valueOf()).to.equal(value);
+        done();
+      });
+
+      it('Number object', function (done) {
+        expect(new Double(new Number(value)).valueOf()).to.equal(value);
+        done();
+      });
+    });
+
+    describe('#toString()', () => {
+      it('should serialize to a string', () => {
+        const testNumber = Math.random() * Number.MAX_VALUE;
+        const double = new Double(testNumber);
+        expect(double.toString()).to.equal(testNumber.toString());
+      });
+
+      const testRadices = [2, 8, 10, 16, 22];
+
+      for (const radix of testRadices) {
+        it(`should support radix argument: ${radix}`, () => {
+          const testNumber = Math.random() * Number.MAX_VALUE;
+          const double = new Double(testNumber);
+          expect(double.toString(radix)).to.equal(testNumber.toString(radix));
+        });
+      }
     });
+  });
+
+  function serializeThenDeserialize(value) {
+    const serializedDouble = BSON.serialize({ d: value });
+    const deserializedDouble = BSON.deserialize(serializedDouble, { promoteValues: false });
+    return deserializedDouble.d;
+  }
+
+  const testCases = [
+    { name: 'Infinity', input: new Double(Infinity) },
+    { name: '-Infinity', input: new Double(-Infinity) },
+    { name: 'Number.EPSILON', input: new Double(Number.EPSILON) },
+    { name: 'Zero', input: new Double(0) },
+    { name: 'Double (Negative Zero)', input: new Double(-0) }
+  ];
 
-    it('Number object', function (done) {
-      expect(new Double(new Number(value)).valueOf()).to.equal(value);
-      done();
+  for (const { name, input } of testCases) {
+    it(`should return Double from serialize-deserialize roundtrip when value is: ${name}`, () => {
+      const outcome = serializeThenDeserialize(input);
+      expect(outcome.value).to.equal(input.value);
+      expect(Object.is(outcome.value, input.value)).to.be.true;
```

Comment:
It may be good to add a comment why we are using two different ways of testing equality here (I assume it's to catch the `0, -0`), but that's not immediately obvious; actually, the equality check isn't necessary if you use `Object.is`, so we can probably get rid of that first assertion.

Reply:
Yeah, the first equality is redundant, I'll remove it.